### PR TITLE
feat(glean): Add front-end glean events for reset pwd with code

### PR DESCRIPTION
**Pull request by vpomerleau**
## Because

* We want to track metrics for reset password flow and measure impact of new design

## This pull request

* Set up new front-end glean metrics for reset password with code:
  * password_reset_email_confirmation_view
  * password_reset_email_confirmation_submit
  * password_reset_email_confirmation_resend_code
  * password_reset_email_confirmation_signin
  * password_reset_email_confirmation_different_account
* and two new additional new events for existing reset password pages:
  * password_reset_recovery_key_cannot_find
  * password_reset_create_new_recovery_key_message_click
* Add Glean pings for theses events in front-end pages

## Issue that this pull request solves

Closes: #FXA-9529

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

### CODE REVIEW

File: packages/fxa-settings/src/components/LinkRememberPassword/index.test.tsx [hunk 1]
```diff
@@ -55,4 +69,20 @@ describe('LinkRememberPassword', () => {
       `/?client_id=123&prefillEmail=${encodeURIComponent(MOCK_EMAIL)}`
     );
   });
+
+  it('renders the expected metrics', async () => {
```
Comment (chenba) [inline packages/fxa-settings/src/components/LinkRememberPassword/index.test.tsx:73]: nit: 'renders' sounds a little odd here

File: packages/fxa-settings/src/components/LinkRememberPassword/index.tsx [hunk 2]
```diff
@@ -25,14 +29,27 @@ const LinkRememberPassword = ({ email }: LinkRememberPasswordProps) => {
     linkHref = params.size > 0 ? `/?${params}` : '/';
   }
 
+  const recordGleanMetric = () => {
+    if (gleanMetric === 'passwordResetEmailConfirmation') {
+      GleanMetrics.passwordReset.emailConfirmationSignin();
+    }
```
Comment (chenba) [inline packages/fxa-settings/src/components/LinkRememberPassword/index.tsx:35]: Interesting.  We can't call the glean function from wherever `gleanMetric` is passed in?

Comment (chenba) [reply packages/fxa-settings/src/components/LinkRememberPassword/index.tsx:35]: Oh I guess the click handler is in here, and we don't accept additional click handling code.

**Review by chenba — APPROVED** [review]
👍

File: packages/fxa-settings/src/components/LinkRememberPassword/index.test.tsx (hunk 1 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/components/LinkRememberPassword/index.test.tsx:73]: Definitely meant for this to be "emit"

File: packages/fxa-settings/src/components/LinkRememberPassword/index.tsx (hunk 2 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/components/LinkRememberPassword/index.tsx:42]: That's a better idea. I've modified the component to take in optional/additional click handling. Thanks!

**Pull request merged**